src/web/dosguard/DOSGuard.cpp

```diff
@@ -115,14 +123,16 @@ DOSGuard::add(std::string const& ip, uint32_t numObjects) noexcept
 }
 
 [[maybe_unused]] bool
-DOSGuard::request(std::string const& ip) noexcept
+DOSGuard::request(std::string const& ip, boost::json::object const& request)
 {
     if (whitelistHandler_.get().isWhiteListed(ip))
         return true;
 
+    auto const weight = weights_.get().requestWeight(request);
+
     {
         auto lock = mtx_.lock<std::scoped_lock>();
-        lock->ipState[ip].requestsCount++;
+        lock->ipState[ip].requestsCount += weight;
```

Comment:
It's not `requestsCount` anymore

Reply:
It is still request count, just some requests cost more than 1 request.